nvm.sh

```diff
@@ -2141,13 +2128,11 @@ nvm_check_file_permissions() {
     setopt nonomatch
   fi
   for FILE in $1/* $1/.[!.]* $1/..?* ; do
-      if [ -d "$FILE" ]; then
-        if ! nvm_check_file_permissions "$FILE"; then
-          if [ "${ZSH_HAS_NONOMATCH_UNSET}" -eq 1 ] && nvm_has "setopt"; then
-            setopt nomatch
-          fi
-          return 2
+      if [ -d "$FILE" ] && ! nvm_check_file_permissions "$FILE"; then
```

Comment:
this isn't the same; if the file is a directory but `nvm_check_file_permissions` returns true, i don't want anything to happen. With this change, it ends up hitting the `else` branch.